File: runtimes/kotlin/src/main/kotlin/io/openruntimes/kotlin/RuntimeContext.kt
package io.openruntimes.kotlin

class RuntimeContext(val req: RuntimeRequest, val res: RuntimeResponse, val logger: RuntimeLogger) {
    fun log(message: Any) {
        this.logger.write(message, RuntimeLogger.TYPE_LOG)
    }

    fun error(message: Any) {
        this.logger.write(message, RuntimeLogger.TYPE_ERROR)
    }
}

File: runtimes/swift/versions/latest/Sources/Runtime/RuntimeContext.swift
import Foundation

protocol CollectionType {}
extension Array: CollectionType {}
extension Set: CollectionType {}
extension Dictionary: CollectionType {}
extension NSArray: CollectionType {}
extension NSSet: CollectionType {}
extension NSDictionary: CollectionType {}

class RuntimeContext: @unchecked Sendable {
    let req: RuntimeRequest
    let res: RuntimeResponse
    let logger: RuntimeLogger

    init(request: RuntimeRequest, response: RuntimeResponse, logger: RuntimeLogger) {
        req = request
        res = response
        self.logger = logger
    }

    func log(_ message: Any) {
        logger.write(message: message, type: RuntimeLogger.TYPE_LOG)
    }

    func error(_ message: Any) {
        logger.write(message: message, type: RuntimeLogger.TYPE_ERROR)
    }
}
